src/uucore/src/lib/mods/posix.rs

```diff
@@ -0,0 +1,52 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore (vars)
+//! Iterate over lines, including the line ending character(s).
+//!
+//! This module provides the [`posix_version`] function, that returns
+//! Some(usize) if the `_POSIX2_VERSION` environment variable is defined
+//! and has value that can be parsed.
+//! Otherwise returns None, so the calling utility would assume default behavior.
+//!
+//! NOTE: GNU (as of v9.4) recognizes three distinct values for POSIX version:
+//! '199209' for POSIX 1003.2-1992, which would define Obsolete mode
+//! '200112' for POSIX 1003.1-2001, which is the minimum version for Traditional mode
+//! '200809' for POSIX 1003.1-2008, which is the minimum version for Modern mode
+//!
+//! Utilities that rely on this module:
+//! `sort` (TBD)
+//! `tail` (TBD)
+//! `touch` (TBD)
+//! `uniq`
+//!
+use std::env;
+
+pub const OBSOLETE: usize = 199209;
```

Comment:
what about using an enum here?

Reply:
yes, initially i tried to implement it as an enum, but discarded in favor of these constants - unfortunately it is inconsistently checked in GNU, so was easier to implement this way for now. Might need to give it another try later on as these will be needed to fully cover GNU tests for at least 2 more utilities

Reply:
ok, thanks